homeassistant/components/daikin_madoka/climate.py

```diff
@@ -0,0 +1,363 @@
+        if self.controller.fan_speed.status is None:
+            return FAN_OFF
+        # pylint: disable=no-else-return
+        if self.hvac_mode == HVAC_MODE_HEAT:
+            return DAIKIN_TO_HA_FAN_MODE.get(
+                self.controller.fan_speed.status.heating_fan_speed
+            )
+        else:
+            return DAIKIN_TO_HA_FAN_MODE.get(
+                self.controller.fan_speed.status.cooling_fan_speed
+            )
+
+    async def async_set_fan_mode(self, fan_mode):
+        """Set fan mode."""
+        try:
+
+            await self.controller.fan_speed.update(
+                FanSpeedStatus(
+                    HA_FAN_MODE_TO_DAIKIN.get(fan_mode),
+                    HA_FAN_MODE_TO_DAIKIN.get(fan_mode),
+                )
+            )
+        except ConnectionAbortedError:
+            # pylint: disable=logging-not-lazy
+            _LOGGER.info(
+                "Could not set target fan mode on %s. "
+                + "Connection not available, please reload integration to try reenabling.",
+                self.name,
+            )
+        except ConnectionException:
+            pass
+
+    @property
+    def fan_modes(self):
+        """List of available fan modes."""
+        return list(HA_FAN_MODE_TO_DAIKIN)
+
+    async def async_update(self):
+        """Retrieve latest state."""
+
+        try:
+            self.dev_info = await self.controller.read_info()
+            await self.controller.update()
+
+        except ConnectionAbortedError:
+            # pylint: disable=logging-not-lazy
+            _LOGGER.info(
+                "Could not update device status for %s. "
+                + "Connection not available, please reload integration to try reenabling.",
+                self.name,
+            )
+        except ConnectionException:
+            pass
+
+    async def async_turn_on(self):
+        """Turn device on."""
+        try:
+            await self.controller.power_state.update(PowerStateStatus(True))
+        except ConnectionAbortedError:
+            # pylint: disable=logging-not-lazy
+            _LOGGER.info(
+                "Could not turn on %s. "
+                + "Connection not available, please reload integration to try reenabling.",
+                self.name,
+            )
+        except ConnectionException:
+            pass
+
+    async def async_turn_off(self):
+        """Turn device off."""
+        try:
+            await self.controller.power_state.update(PowerStateStatus(False))
+        except ConnectionAbortedError:
+            # pylint: disable=logging-not-lazy
+            _LOGGER.info(
+                "Could not turn off %s. "
+                + "Connection not available, please reload integration to try reenabling.",
+                self.name,
+            )
+        except ConnectionException:
+            pass
+
+    @property
+    def device_info(self):
+        """Return a device description for device registry."""
+
+        return self.dev_info
```

Comment:
No, the properties may no be fully aligned. When I reviewed what was being done with them I just thought it would do no harm to let HA list only those that match. I will change it to the supported properties as suggested.